### PR TITLE
Configure ports

### LINKED ISSUE #25: Configure InfluxDB Address and Port
**Is your feature request related to a problem? Please describe.**
localhost:8089 is hardcoded into the InfluxDB V1 handler. 

**Describe the solution you'd like**
Update the code to take Zap config. Update the default GSW config to have configuration values for it where they are localhost and 8089

### PR DESCRIPTION
**Pull request by Smartinez3324**
This closes #25 

# Description
Added hostname and port number configuration using viper flags, default values will be localhost and 8089.

# How Has This Been Tested?
Tested changing the values of the host name and port numbers using the flags I created. checked that they were parsed by net.ResolveUDPAddr 

# Checklist:
- [x] New functionality is documented in the necessary spots (i.e new functions documented in the header)
- [x] Unit tests cover any new functionality or edge cases that the PR was meant to resolve (if applicable) 
- [x] The CI checks are passing
- [x] I reviewed my own code in the GitHub diff and am sure that each change is intentional

### CODE REVIEW

File: cmd/gsw_service.go [hunk 1]
```diff
@@ -114,10 +114,14 @@ func dbInitialize(ctx context.Context, channelMap map[int]chan []byte) error {
 func readConfig() *viper.Viper {
 	config := viper.New()
 	configFilepath := flag.String("c", "gsw_service", "name of config file")
+	databasePortNumber := flag.Int("dp", 8089, "port number for the database")
+	databaseHostName := flag.String("dh", "localhost", "database host name")
 	flag.Parse()
 	config.SetConfigName(*configFilepath)
 	config.SetConfigType("yaml")
 	config.AddConfigPath("data/config/")
+	config.Set("databasePortNumber", databasePortNumber)
```
Comment (apenza) [inline cmd/gsw_service.go:123]: It seems like this overrides the values in the config file even if the flags aren't set (using the default flag values). Maybe get rid of the default flag values and only override if the flags are actually set?

File: cmd/gsw_service.go [hunk 2]
```diff
@@ -114,10 +114,14 @@ func dbInitialize(ctx context.Context, channelMap map[int]chan []byte) error {
...
+	config.Set("databaseHostName", databaseHostName)
```
Comment (apenza) [inline cmd/gsw_service.go:124]: Consider using underscores (i.e. database_port_number and database_host_name) rather than camel case for consistency with what is already in the config file.

File: cmd/gsw_service.go (hunk 1 above)
Comment (apenza) [reply cmd/gsw_service.go:123]: I think we should add these values to the gsw_config file so that we don't have to use the flags every time if we want to change the host / port.

**Review by apenza — CHANGES_REQUESTED** [review]

File: cmd/gsw_service.go (hunk 1 above)
Comment (AarC10) [reply cmd/gsw_service.go:123]: ^ Upvote on this

File: cmd/gsw_service.go (hunk 2 above)
Comment (AarC10) [reply cmd/gsw_service.go:124]: Also another upvote

**Review by AarC10 — CHANGES_REQUESTED** [review]

File: cmd/gsw_service.go (hunk 2 above)
Comment (Smartinez3324) [reply cmd/gsw_service.go]: Done

File: data/config/gsw_service.yaml [hunk 3]
```diff
@@ -1,2 +1,4 @@
 telemetry_config: data/config/backplane.yaml
 logging_config: data/config/logger.yaml
+database_port_number:
+database_host_name:
```
Comment (AarC10) [inline data/config/gsw_service.yaml:4]: Put the default here too.

**Review by AarC10 — CHANGES_REQUESTED** [review]

File: data/config/gsw_service.yaml (hunk 3 above)
Comment (Smartinez3324) [reply data/config/gsw_service.yaml:4]: ok

File: cmd/gsw_service.go [hunk 4]
```diff
@@ -115,6 +115,8 @@ func readConfig() *viper.Viper {
 	config := viper.New()
 	configFilepath := flag.String("c", "gsw_service", "name of config file")
 	flag.Parse()
+	config.SetDefault("database_port_number", 8089)
+	config.SetDefault("database_host_name", "localhost")
```
Comment (AarC10) [inline cmd/gsw_service.go:119]: Blast these. I assume theres an error that gets thrown if the key isnt found. The user should have this set which is done in the default config

**Review by AarC10 — CHANGES_REQUESTED** [review]

File: cmd/gsw_service.go (hunk 4 above)
Comment (Smartinez3324) [reply cmd/gsw_service.go:119]: 👍

Comment (Smartinez3324) [reply cmd/gsw_service.go:119]: if nothing is found it does not throw an error, do you want it to? and if it doesn't find anything what do you want the program to do?

Comment (AarC10) [reply cmd/gsw_service.go:119]: Yea lets throw an error. By default it wont error if we give it this default file, so that means the user would've had to make an incorrect modification. Debating on the last part. Just exit early i would say now is fine since we do probably want to log our data no matter what, but in the future we configure to allow disabling it

Comment (Smartinez3324) [reply cmd/gsw_service.go:119]: so for now, you want it to error if it doesn't detect a database_port_number and database_host_name, and later on you want a setting where you can turn off sending data to the database?

**Review by AarC10 — APPROVED** [review]

**Review by apenza — APPROVED** [review]

**Review by Smartinez3324 — COMMENTED** [review]
I reviewed this :+1:

**Review by AarC10 — APPROVED** [review]

**Comment by AarC10** [comment]
MURGE
